internal/pkg/service/buffer/api/dependencies/dependencies.go

```diff
@@ -111,11 +111,15 @@ type forProjectRequest struct {
 
 func NewServerDeps(serverCtx context.Context, envs env.Provider, logger log.PrefixLogger, debug, dumpHttp bool) (v ForServer, err error) {
 	// Create tracer
+	ctx := serverCtx
 	var tracer trace.Tracer = nil
 	if telemetry.IsDataDogEnabled(envs) {
+		var span trace.Span
 		tracer = telemetry.NewDataDogTracer()
-		_, span := tracer.Start(serverCtx, "kac.lib.api.server.buffer.dependencies.NewServerDeps")
+		ctx, span = tracer.Start(ctx, "kac.lib.api.server.buffer.dependencies.NewServerDeps")
 		defer telemetry.EndSpan(span, &err)
+	} else {
+		tracer = telemetry.NewNopTracer()
```

Comment:
Pridal som `telemetry.NewNopTracer()`, kedze Honza na to narazil, ze inak je tu `nil` hodnota.
Ono sa to sice riesilo, ale az dalej v `base deps`.

Reply:
span `ctx` pouzijem neskor v dalsom commite